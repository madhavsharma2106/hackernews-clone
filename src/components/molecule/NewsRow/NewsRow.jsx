import React from "react";
import Text, { TextVariant } from "../../atoms/Text/Text";
import { calculateTimeLapsed, extractHostNameFromUrl } from "../../../utils";
import "./NewsRow.scss";

function NewsRow({ id, votes, row, addVote, hideRow }) {
  return (
    <tr>
      <td>
        <Text variant={TextVariant.regular}>{row.num_comments}</Text>
      </td>

      <td>
        <Text variant={TextVariant.regular}>{votes ?? 0}</Text>
      </td>
      <td>
        <Text
          aria-label="Upvote"
          variant={TextVariant.regular}
          clickable
          onClick={() => addVote(id)}
        >
          &#8679;
        </Text>
      </td>
      <td className="news-details">
        <Text variant={TextVariant.regular} inline>
          {row.title}
        </Text>
        <Text
          aria-label="Link to article"
          variant={TextVariant.subText}
          inline
          clickable
          onClick={() => {
            window.open(row.url, "_blank");
          }}
        >
          {row.url && extractHostNameFromUrl(row.url)}
        </Text>
        <Text variant={TextVariant.subText} inline>
          by
        </Text>
        <Text variant={TextVariant.subTextDark} inline>
          {row.author}
        </Text>
        <Text variant={TextVariant.subText} inline>
          {`${calculateTimeLapsed(row.created_at)} ago`}
        </Text>
        <Text
          variant={TextVariant.subTextDark}
          inline
          clickable
          onClick={() => hideRow(id)}
        >
          hide
        </Text>
      </td>
    </tr>
  );
}

export default NewsRow;
