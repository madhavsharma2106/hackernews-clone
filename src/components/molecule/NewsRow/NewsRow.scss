@import "../../styles/main.scss";

tbody {
  tr {
    background: $primary-brown;
  }
  tr:nth-child(even) {
    background: $secondary-brown;
  }
}
.news-details {
  p {
    margin: auto 2px;
  }
}
