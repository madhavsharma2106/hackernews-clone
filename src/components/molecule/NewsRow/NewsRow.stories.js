import React from "react";
import { withKnobs } from "@storybook/addon-knobs";
import { withInfo } from "@storybook/addon-info";
import NewsRow from "./NewsRow";

export default {
  component: NewsRow,
  title: "Molecules/NewsRow",
  decorators: [withKnobs, withInfo],
  parameters: {
    info: { inline: true, header: false, source: true },
  },
};

export const Sample = () => {
  const row = {
    num_comments: 1,
    url: "http://random.com",
    title: "test title",
    author: "test author",
    created_at: "2020-05-23T08:00:07.000Z",
  };
  return <NewsRow id="123" votes={123} row={row} />;
};
