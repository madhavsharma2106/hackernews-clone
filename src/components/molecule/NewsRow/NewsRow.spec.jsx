import { shallow } from "enzyme";
import React from "react";
import NewsRow from "./NewsRow";

const row = {
  num_comments: 1,
  url: "http://random.com",
  title: "test title",
  author: "test author",
};

describe("News Row Component", () => {
  it("should render row without errors", () => {
    const wrapper = shallow(
      <NewsRow
        id="123"
        votes={123}
        row={row}
        addVote={() => {}}
        hideRow={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
