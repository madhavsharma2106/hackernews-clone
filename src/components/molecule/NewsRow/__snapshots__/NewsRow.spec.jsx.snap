// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Row Component should render row without errors 1`] = `
<tr>
  <td>
    <Text
      variant="regular"
    >
      1
    </Text>
  </td>
  <td>
    <Text
      variant="regular"
    >
      123
    </Text>
  </td>
  <td>
    <Text
      aria-label="Upvote"
      clickable={true}
      onClick={[Function]}
      variant="regular"
    >
      â‡§
    </Text>
  </td>
  <td
    className="news-details"
  >
    <Text
      inline={true}
      variant="regular"
    >
      test title
    </Text>
    <Text
      aria-label="Link to article"
      clickable={true}
      inline={true}
      onClick={[Function]}
      variant="sub"
    >
      (random.com)
    </Text>
    <Text
      inline={true}
      variant="sub"
    >
      by
    </Text>
    <Text
      inline={true}
      variant="sub-text-dark"
    >
      test author
    </Text>
    <Text
      inline={true}
      variant="sub"
    >
      0 hours ago
    </Text>
    <Text
      clickable={true}
      inline={true}
      onClick={[Function]}
      variant="sub-text-dark"
    >
      hide
    </Text>
  </td>
</tr>
`;
